src/mrsimulator/method/tests/test_query.py

```diff
@@ -87,26 +92,44 @@ def test_MixingQuery():
     assert obj1.channels == [rf1, rf2, None]
 
     # direct initialization
-    test2 = {"ch2": {"angle": 5.101, "phase": 1.61}}
+    test2 = {"ch2": {"angle": 1.101, "phase": 1.61}}
     obj2 = MixingQuery(**test2)
     assert obj2.name is None
     assert obj2.description is None
     assert obj2.label is None
     assert obj2.ch1 is None
-    assert obj2.ch2 == RotationQuery(angle=5.101, phase=1.61)
+    assert obj2.ch2 == RotationQuery(angle=1.101, phase=1.61)
     assert obj2.ch3 is None
 
+    # JSON tests
+    test3 = {"ch1": {"angle": 1.23, "phase": 1.23}}
+    obj3 = MixingQuery(**test3)
+    assert obj3.json(units=False) == test3
+    assert obj3.json(units=True) == {"ch1": {"angle": "1.23 rad", "phase": "1.23 rad"}}
+
 
 def test_MixingEnum():
     total_mix = MixingEnum["TotalMixing"]
     assert total_mix.value == "TotalMixing"
+    assert total_mix.json(units=True) == "TotalMixing"
+    assert total_mix.json(units=False) == "TotalMixing"
```

Comment:
Are we serializing `TotalMixing`? I think if mixing is not defined between two spectral/delay events it is assumed as total mixing. Not sure what would happen when you explicitly define `TotalMixing`. Can you check this through a simulation---both in a Python script and when loading serialized JSON file?

Reply:
When no `MixingEvent` is present between two spectral/delay events, then the default is to connect all transitions, but no `MixingEnum.TotalMixing` is present. Serializing this method would not include `"TotalMixing"` in the JSON.

Writing the equivalent method, but including `TotalMixing` (for clarity, no other benefit) will produce the same simulated spectra, and the method seralization will include `"TotalMixing"` in the JSON.

See the attached notebook
[MixingEnum_example.ipynb.zip](https://github.com/deepanshs/mrsimulator/files/11778724/MixingEnum_example.ipynb.zip)

Reply:
I found two issues with this:
1. **When reading the method from a dict.**
```py
mth = Method.parse_dict_with_units(threeQ_MAS_total_mixing.json())
```
gives
```
BaseException: 'inf Hz' did not parse as unit: At col 0, inf is not a valid unit.
```
We should replace np.inf with 1e12 when serializing and add a dedicated test.

2. **When reading the method from a dict after substituting `np.inf` for 1e12 in the method.**
```py
mth = Method.parse_dict_with_units(threeQ_MAS_total_mixing.json())
```
gives
```
ValidationError: 1 validation error for SpectralEvent
query
  extra fields not permitted (type=value_error.extra)
```

The parser only checks for `SpectralEvent` or `DelayEvent`. We don't have any parse for `MixingEvent`. We need to write one and have a dedicated test for that.

Reply:
Another issue with the event parser is the inability to differentiate between a `SpectralEvent` and `DelayEvent` with default `fraction`/`delay` attributes. At the time of this comment, `DelayEvent` objects have not been merged into master, but still something to think about.

Reply:
@deepanshs A fix for serializing/loading inf rotor frequency would be adding a validator for `rotor_frequency` which converts an infinite speed to `10e12 Hz` for all methods.

This is currently done for 2D named methods (PR #231), but not custom Method objects. Thoughts?

Reply:
> Another issue with the event parser is the inability to differentiate between a `SpectralEvent` and `DelayEvent` with default `fraction`/`delay` attributes. At the time of this comment, `DelayEvent` objects have not been merged into master, but still something to think about.

We can add a type key to objects. For `SpectraEvent -> {"type": "spectral"}`, for `DelayEvent -> {"type": "delay"}`, and `MixingEvent -> {"type": "mixing"}` which should be a required key and always gets serialized. type doesn't need to be specified when initializing from objects, for example, `SpectralEvent(fraction=0.5)` is still valid, but internally the object should create a type key.

Reply:
Maybe we should wait till the delay event PR is merged.

Reply:
Will address these issues now that DelayEvent has been implemented

Reply:
I worked through some ideas for implementing a `event_type` keyword in serialized JSON for events, although none of them integrate smoothly into the current code. I would like to move the improved parsing for events into another PR and merge this one.